contrib/format-esri/src/main/java/org/apache/drill/exec/store/esri/ShpBatchReader.java

```diff
@@ -0,0 +1,334 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ShpBatchReader implements ManagedReader<FileSchemaNegotiator> {
+
+  private FileSplit split;
+  private BufferedReader reader;
+  private ResultSetLoader loader;
+  private ShpReaderConfig readerConfig;
+  private Path hadoopShp;
+  private Path hadoopDbf;
+  private Path hadoopPrj;
+  private FSDataInputStream fileReaderShp = null;
+  private FSDataInputStream fileReaderDbf = null;
+  private FSDataInputStream fileReaderPrj = null;
+  private GeometryCursor geomCursor = null;
+  private DbfReader dbfReader = null;
+  private ScalarWriter gidWriter;
+  private ScalarWriter sridWriter;
+  private ScalarWriter shapeTypeWriter;
+  private ScalarWriter geomWriter;
+  private RowSetLoader rowWriter;
+
+
+  private int srid;
+  private SpatialReference spatialReference;
+  private static final Logger logger = LoggerFactory.getLogger(ShpBatchReader.class);
+
+  public static class ShpReaderConfig {
+    protected final ShpFormatPlugin plugin;
+
+    public ShpReaderConfig(ShpFormatPlugin plugin) {
+      this.plugin = plugin;
+    }
+  }
+
+  public ShpBatchReader(ShpReaderConfig readerConfig) {
+    this.readerConfig = readerConfig;
+  }
+
+  @Override
+  public boolean open(FileSchemaNegotiator negotiator) {
+    this.split = negotiator.split();
+    this.hadoopShp = split.getPath();
+    this.hadoopDbf = new Path(split.getPath().toString().replace("shp", "dbf"));
+    this.hadoopPrj = new Path(split.getPath().toString().replace("shp", "prj"));
+
+    openFile(negotiator);
+    SchemaBuilder builder = new SchemaBuilder();
+    builder.addNullable("gid", TypeProtos.MinorType.INT);
+    builder.addNullable("srid", TypeProtos.MinorType.INT);
+    builder.addNullable("shapeType", TypeProtos.MinorType.VARCHAR);
+    builder.addNullable("geom", TypeProtos.MinorType.VARBINARY);
+
+    negotiator.setTableSchema(builder.buildSchema(), false);
+    loader = negotiator.build();
+
+    rowWriter = loader.writer();
+    gidWriter = rowWriter.scalar("gid");
+    sridWriter = rowWriter.scalar("srid");
+    shapeTypeWriter = rowWriter.scalar("shapeType");
+    geomWriter = rowWriter.scalar("geom");
+
+    return true;
+  }
+
+  @Override
+  public boolean next() {
+    if (fileReaderShp == null) {
+      return false;
+    }
+
+    Geometry geom = null;
+
+    while( !rowWriter.isFull()){
+      Object[] dbfRow = dbfReader.nextRecord();
+      geom = geomCursor.next();
+      if(geom == null) {
+        return false;
+      }
+      processShapefileSet(rowWriter, geomCursor.getGeometryID(), geom, dbfRow);
+    }
+    return true;
+  }
+
+  private void openFile(FileSchemaNegotiator negotiator) {
+    try {
+      fileReaderShp = negotiator.fileSystem().open(split.getPath());
```

Comment:
Since format can be compressible, `openPossiblyCompressedStream` should be used.

Reply:
Thanks and included